extensions/2.0/Vendor/EXT_feature_metadata/0.0.0/README.md

```diff
@@ -0,0 +1,665 @@
+
+**Note**: Since glTF accessors do not support `UNSIGNED_INT` types, `FLOAT32` must be used instead. This allows for integer feature IDs up to `2e24`.
+
+Once the attribute is defined, it must be associated with a primitive. This is done by setting `featureIds.attribute` to reference the `_FEATURE_ID_X`. See the following code snippet for an example of this.
+
+```json
+{
+  "extensions": {
+    "EXT_feature_metadata": {
+      "classes": {
+        "room": {
+          "name": "Room",
+          "description": "room of the house",
+          "properties": {
+            "name": {
+              "description": "Name of the room such as 'kitchen' or 'living room'", 
+              "type": "STRING"
+            },
+            "wallColor": {
+              "name": "Wall Color",
+              "type": "STRING"
+            },
+            "area": {
+              "name": "Area (m^2)",
+              "type": "FLOAT32"
+            },
+            "furniture": {
+              "name": "Furniture",
+              "description": "List of furniture in the room such as 'chair' or 'table'",
+              "type": "ARRAY",
+              "componentType": "STRING"
+            }
+          }
+        }
+      },
+      "featureTables": {
+        "roomTable": {
+          "class": "room",
+          "count": 8,
+          "properties": {
+            "name": {
+              "bufferView": 4,
+              "offsetBufferViews": [5]
+            },
+            "wallColor": {
+              "bufferView": 6,
+              "offsetBufferViews": [7]
+            },
+            "area": {
+              "bufferView": 8,
+            },
+            "furniture": {
+              "bufferView": 9,
+              "offsetBufferViews": [10, 11]
+            }
+          }
+        }
+      }
+    }
+  },
+  "meshes": [
+    {
+      "name": "Point Cloud Mesh",
+      "primitives": [
+        {
+          "attributes": {
+            "POSITION": 0,
+            "_FEATURE_ID_0": 1
+          },
+          "mode": 0,
+          "extensions": {
+            "EXT_feature_metadata": {
+              "featureIdAttributes": [
+                {
+                  "featureTable": "weatherTable",
+                  "featureIds": {
+                    "constant": 0,
+                    "vertexStride": 1
+                  }
+                },
+                {
+                  "featureTable": "townTable",
+                  "featureIds": {
+                    "attribute": "_FEATURE_ID_0"
+                  }
+                }
+              ]
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Feature ID Mappings
+
+![Feature ID Mappings diagram](figures/feature-id-mappings.jpg)
+
+For large datasets, explicitly listing every feature ID can be expensive. To help avoid this, **feature ID mappings** allow for an implicit feature ID numbering scheme for common cases:
```

Comment:
Something doesn't quite seem right with the name "Feature ID Mappings". What is mapping between? Vertex ID and Feature ID? Maybe that should be explicitly stated.

Reply:
good point. yes, it's a mapping from `vertex ID -> feature ID`